lib/utils/net-utils.js

```diff
@@ -0,0 +1,25 @@
+const net = require('net')
+
+const NetUtils = {
+  isPortAvailable (port) {
+    return new Promise((resolve) => {
+      const server = net.createServer()
+
+      server.unref()
+      server.on('error', () => resolve(false))
```

Comment:
I suppose we have to check `EADDRINUSE`, `EACCES` codes, otherwise reject the promise and terminate karma:
```js
.on('error', function(e) {
    if (e.code != 'EADDRINUSE' && e.code!='EACCES'){
        return reject(e);
    }
    resolve(false)
```

Reply:
Do you mean any specific situation when such check would be relevant ?

Reply:
As an example you could specify invalid `listenAddress` in karma.config: `listenAddress: '123'`. It should fail with `EADDRNOTAVAIL` error.

Reply:
So, I suppose it would be safer to limit checks using known errors related to port usage.

Reply:
@sergei-startsev 
I've updated pull request with your suggestion. What do you think about current solution ?

Reply:
@lusarz thanks for the updates, I've left a few more comments that might be useful.